src/shared/sam/cli/samCliInvoker.ts

```diff
@@ -35,6 +35,8 @@ export class DefaultSamCliProcessInvoker implements SamCliProcessInvoker {
         let options: SpawnOptions | undefined
         if (Array.isArray(optionsOrArgs)) {
             args = optionsOrArgs as string[]
+        } else if (typeof optionsOrArgs === 'string' || optionsOrArgs instanceof String) {
```

Comment:
This doesn't seem correct...

1. `optionsOrArgs` is of type `SpawnOptions` or `string[]`, never `string`.
2. `instanceof` is redundant, and will cause an error if `optionsOrArgs` is undefined.
3. If `optionsOrArgs` *could* be a string, the compiler is aware of `typeof` causes for build-in types, so you wouldn't need to call `optionsOrArgs.toString()`

I'll put together a PR with the fix.